single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/library/ResourcesComponentLibrary.java

```diff
@@ -170,8 +174,25 @@ public Optional<String> getSubComponentStyleClass(String type, String subCompone
     @Override
     public Map<Orientation, Component.Transformation> getTransformations(String type) {
         Objects.requireNonNull(type);
-        Component component = components.get(type);
+        Component component = getComponent(type);
         return component != null ? component.getTransformations() : Collections.emptyMap();
     }
 
+    protected Component getComponent(String type) {
+        Objects.requireNonNull(type);
+        Component component = components.get(type);
+        if (component == null && isInsideVlComponent(type)) {
+            component = components.get(ComponentTypeName.UNKNOWN_COMPONENT);
+        }
+        return component;
+    }
+
+    protected boolean isInsideVlComponent(String componentType) {
```

Comment:
Could you rename this to `isDisplayed`?
I think we should address this in another way later : adding the "invisible" components in the `components.json`.